core/templates/dev/head/App.ts

```diff
@@ -103,12 +104,13 @@ oppia.config([
     // Add an interceptor to convert requests to strings and to log and show
     // warnings for error responses.
     $httpProvider.interceptors.push([
-      '$q', '$log', 'AlertsService', function($q, $log, AlertsService) {
+      '$q', '$log', 'AlertsService', 'CsrfTokenService',
+      function($q, $log, AlertsService, CsrfTokenService) {
         return {
           request: function(config) {
             if (config.data) {
               config.data = $.param({
-                csrf_token: GLOBALS.csrf_token,
+                csrf_token: CsrfTokenService.getToken(),
```

Comment:
Currently, if it happens that the token is still null, an error will pop up telling the user to report this bug or to refresh the page

Reply:
This should be solvable by Promises

Reply:
Done